### PR TITLE
mesh: add xRoute ACL hook tenancy tests

### PR DESCRIPTION
**Pull request by rboyer**
### Description

Enhance the xRoute ACL hook tests to cover tenanted situations. These tests will only execute in enterprise.

### CODE REVIEW

File: internal/mesh/internal/types/xroute_test.go [hunk 1]
```diff
@@ -457,19 +473,29 @@ func testXRouteACLs[R XRouteData](t *testing.T, newRoute func(t *testing.T, pare
...
-			checkF(t, "write", tc.writeOK, reg.ACLs.Write(authz, &acl.AuthorizerContext{}, tc.res))
-			checkF(t, "list", DEFAULT, reg.ACLs.List(authz, &acl.AuthorizerContext{}))
+			checkF(t, "read", tc.readOK, reg.ACLs.Read(authz, authCtx, tc.res.Id, tc.res))
+			checkF(t, "write", tc.writeOK, reg.ACLs.Write(authz, authCtx, tc.res))
+			checkF(t, "list", DEFAULT, reg.ACLs.List(authz, authCtx))
 		})
 	}
 
-	serviceRead := func(name string) string {
+	isEnterprise := (structs.NodeEnterpriseMetaInDefaultPartition().PartitionOrEmpty() == "default")
```
Comment (ishustava) [inline internal/mesh/internal/types/xroute_test.go:487]: What if you are running in enterprise but in the `default` partition? In that case, we won't be testing the partitioned policy for the default partition it seems

Comment (rboyer) [reply internal/mesh/internal/types/xroute_test.go:487]: This specific call to `PartitionOrEmpty()` returns the following:
- CE: `""`
- ENT: `"default"`

So as long as the build was produced using `-tags consulent` then `isEnterprise` will always be `true`.

Comment (ishustava) [reply internal/mesh/internal/types/xroute_test.go:487]: Ah ok! My question was more about if both builds return the same value, but in this case that's not a problem.

**Review by ishustava — APPROVED** [review]

File: internal/mesh/internal/types/xroute_test.go (hunk 1 above)
Comment (rboyer) [reply internal/mesh/internal/types/xroute_test.go:487]: I'll replace this hack with https://github.com/hashicorp/consul/pull/19201 to make it more obvious what's going on.

**Pull request merged**